### ISSUE TITLE
Can you request a single cert with a wildcard and the base domain with our DNS plugins

### ISSUE THREAD

**Issue by bmw** [certbot/certbot#5472]
There was some concern from @joohoi about whether or not we currently handle this correctly in any/all of our DNS plugins. @joohoi, if you could add some additional context here that'd be great. 

We should look into this issue soon as we're currently in the process of getting our DNS plugins packaged everywhere and we should try to avoid making distros immediately repackage them.

**Comment by joohoi** [comment]
Yeah, for this to work, we would need to add two `TXT` records to a single subdomain in the DNS zone, ie: your certificate needs names `yourdomain.tld` and `*.yourdomain.tld`, the record that the CA will check is same for both of these: `_acme-challenge.yourdomain.tld` hence a query to that subdomain should return two `TXT` records.

The actual credit for raising this issue goes to @sahsanu for raising the issue up in another context.

**Comment by joohoi** [comment]
Our DNS plugins do this intelligently, and everything should be in order. This includes cleanup as well, the records are cleaned based on their actual content.

**Comment by bmw** [comment]
I just hit this problem when using the Google DNS plugin when testing #5620. Log is: 
[letsencrypt.log.txt](https://github.com/certbot/certbot/files/1753963/letsencrypt.log.txt)

**Comment by trinopoty** [comment]
I think it's better to keep the two TXT records for with and without wildcard of different name. Changing the name of the wildcarded version to something like `_acme-challenge-wildcard` would indicate to the user that it's for a wildcard domain and keep problems like these (with Google DNS, etc) to a minimum.

**Comment by joohoi** [comment]
> I just hit this problem when using the Google DNS plugin when testing

I originally thought this was a limitation of Google Cloud DNS API, but it turns out that Google _does_ in fact support adding multiple TXT records with the same name, they just have to be created in a single API request. This conflicts with the way that DNS plugins currently handle the record creation.

With our DNS plugin API, the only way to manage this would be to first read TXT records for the validation domain, then deleting the old one(s), and finally writing the list of pre-existing TXT records with the addition of the current validation token. The API call for addition takes a list of strings, which in our case would of course be the list of validation tokens. This is done here https://github.com/certbot/certbot/blob/master/certbot-dns-google/certbot_dns_google/dns_google.py#L117

**Comment by bmw** [comment]
> With our DNS plugin API, the only way to manage this would be to first read TXT records for the validation domain, then deleting the old one(s), and finally writing the list of pre-existing TXT records with the addition of the current validation token.

I think we should probably do that.

I also probably think we should go through each of our DNS plugins and test there's not a problem here. Do you have the cycles to help with either of those things this week?

**Comment by ohemorange** [comment]
Checklist @ https://github.com/certbot/certbot/issues/5667

**Comment by bmw** [comment]
Once a PR for route53 lands, this issue can be closed. `lexicon` doesn't have the support we need it to to solve this so I created an issue on their repo and opened #5673 to track the general issue for our lexicon based plugins.

**Comment by bmw** [comment]
Closing in favor of the lexicon specific issue.
